### PR TITLE
tweak(client/cl_spectate): Make gamer tag tickless

### PR DESCRIPTION
**Pull request by Ekinoxx0**
Make cl_spectate tickless

This was kinda bothering me while profiling my server that doesn't use monitor, so I made this part of the code tickless

### CODE REVIEW

**Review by TasoOneAsia — APPROVED** [review]

**Comment by tabarra** [comment]
@TasoOneAsia  b-but... conflict with your own pr

**Comment by Ekinoxx0** [comment]
I could re-do/change this PR if needed after his commit is merged first, just need to know if something is blocking the merge of #591 🤔

**Comment by tabarra** [comment]
Right now merging any client side lua is very risky because errors wont be printed, so we will have really hard time finding out and fixing issues.  
It will take a few more months until we can merge that.

**Comment by Ekinoxx0** [comment]
Seems understandable, implementing some sort of trycatch functions could be a solution with error handling being reported through a sentry. I suppose it would need approval from the Cfx team

Anyway I will keep following both PR and modifying them if needed 👍

**Review by tekkenkkk — APPROVED** [review]
looks good 👍 got it already on my server

**Comment by tabarra** [comment]
Thanks for the PR.  
Due to conflicts I'm merging it manually.

**Pull request closed without merging**
